### PR TITLE
miscellaneous Maverick tweaks

### PR DESCRIPTION
**Pull request by yulolimum**
## Please verify the following:

- [x] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [x] `README.md` has been updated with your changes, if relevant

## Describe your PR

- There was a recent change made to TextField that made it inconsistent with the design. It somewhat broke right/left accessory default alignment as well. Adds tweaks to incorporate the change but make it inline with design.
- Adds setting to hide TabBar when keyboard opens in DemoNavigator.
- The RNR screen was causing crash. Tracked it to the passing `NaN` to the date-fns parsing function. Changes to parse using date-fns and wraps in try..catch.
- Adds podcast icon to RNR screen.

## Proofs

| RNR Screen Icon | TextField Tweaks |
|---|---|
| yulolimum-capture-2022-08-25--22-08-09@2x | yulolimum-capture-2022-08-25--22-08-44@2x |
|  |  |
|  |  |
|  |  |
|  |  |
|  |  |

### CODE REVIEW

File: docs/Theming.md
Comment (yulolimum) [inline docs/Theming.md:3]: linter/formatter did this. not sure if matters

**Review by mazenchami — APPROVED** [review]
This PR looks good to me. I went ahead and create a new [PR](https://github.com/infinitered/ignite/pull/2067) with this as its base that makes some updates for styling and dates to make sure we're accounting for i18n

File: boilerplate/app/components/Icon.tsx
Comment (robinheinze) [inline boilerplate/app/components/Icon.tsx:100]: ❤️ Love this icon!

File: docs/Theming.md
Comment (robinheinze) [reply docs/Theming.md:3]: Yeah it did that to me in #2037 as well

File: boilerplate/app/components/TextField.tsx
Comment (robinheinze) [inline boilerplate/app/components/TextField.tsx:253]: Any reason not to use the spacing scale throughout?

File: boilerplate/app/models/Episode.ts
Comment (robinheinze) [inline boilerplate/app/models/Episode.ts:34]: 👍

**Review by robinheinze — COMMENTED** [review]
Just a question about spacing. Otherwise LGTM 👍

File: boilerplate/app/components/TextField.tsx
Comment (kateinkim) [reply boilerplate/app/components/TextField.tsx:253]: I will be updating this throughout the app starting with demo screens https://github.com/infinitered/ignite/issues/2068 @robinheinze

Comment (mazenchami) [reply boilerplate/app/components/TextField.tsx:253]: @kateinkim please take a look at #2067 since I handle some RTL styling there. that PR gets merged here, so when these two are merged you should be good to go

Comment (kateinkim) [reply boilerplate/app/components/TextField.tsx:253]: Yep I will be waiting on it until it's merged! Thanks @mazenchami

Comment (yulolimum) [reply boilerplate/app/components/TextField.tsx]: > Any reason not to use the spacing scale throughout?

@robinheinze no reason, just creating more work for y'all! :) tbh though, i just haven't done it since we had the array thing. @kateinkim do you want me to update this PR with the spacing or you'll be going over the entire repo changing this?

Comment (kateinkim) [reply boilerplate/app/components/TextField.tsx]: I can go over the entire repo :)

**Pull request merged**
